app/controllers/my_controller.rb

```diff
@@ -78,7 +78,27 @@ def change_password
   end
 
   # Administer access tokens
-  def access_token; end
+  def access_token
+    @storage_tokens = OAuthClientToken
+                        .preload(:oauth_client)
+                        .joins(:oauth_client)
+                        .where(user: @user, oauth_client: { integration_type: 'Storages::Storage' })
```

Comment:
does this match for `integration_type: Storages::NextcloudStorage`? I'm not sure, what type we enter here. But if someone changes that in future, will it be a problem? Just asking out of curiosity, doesn't think this line needs a change.

Reply:
Good question!

⚠️ I'm citing this out of the top of my mind, so details might be a little off ⚠️ 

`Storages::NextcloudStorage` is implemented as a STI on `Storages::Storage`, meaning that to for the purposes of associations it is a `Storage::Storage`.

If and when we load the association, though, the STI mechanism will instantiate the actual `Storages::SpecializedStorage`.

You can see it in place by running `OAuthClientToken.first.oauth_client.integration`